openebs/pkg/v1/snapshot_api.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/golang/glog"
+	mayav1 "github.com/kubernetes-incubator/external-storage/openebs/types/v1"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// CreateSnapshot to create the Vsm through a API call to m-apiserver
+func (v OpenEBSVolume) CreateSnapshot(volName string, snapName string) (string, error) {
+	addr := os.Getenv("MAPI_ADDR")
+	if addr == "" {
+		err := errors.New("MAPI_ADDR environment variable not set")
+		glog.Errorf("Error getting maya-apiserver IP Address: %v", err)
+		return "Error getting maya-apiserver IP Address", err
+	}
+
+	var snap mayav1.SnapshotAPISpec
+
+	snap.Kind = "VolumeSnapshot"
+	snap.APIVersion = "v1"
+	snap.Metadata.Name = snapName
+	snap.Spec.VolumeName = volName
+
+	url := addr + "/latest/snapshots/create/"
+
+	//Marshal serializes the value provided into a YAML document
+	yamlValue, _ := yaml.Marshal(snap)
+
+	glog.V(2).Infof("[DEBUG] snapshot Spec Created:\n%v\n", string(yamlValue))
```

Comment:
Is it a good practice to use [DEBUG]?

Reply:
I will clean this up , and set proper log levels